server/channels/web/oauth.go

```diff
@@ -339,7 +339,45 @@ func completeOAuth(c *Context, w http.ResponseWriter, r *http.Request) {
 	} else if action == model.OAuthActionSSOToEmail {
 		redirectURL = app.GetProtocol(r) + "://" + r.Host + "/claim?email=" + url.QueryEscape(props["email"])
 	} else {
-		session, err := c.App.DoLogin(c.AppContext, w, r, user, "", isMobile, false, false)
+		desktopToken := ""
```

Comment:
This function is already really large. Can you break up this logic into one or more helper functions? That will also make it easier to write tests for it.

Reply:
> This function is already quite large. Can you break up this logic into one or more helper functions? That will make it more readable also make it easier to write tests for.

I agree that the function is too large but this logic was already inside the function, I just moved it further up. I'm happy to refactor this in a separate PR but I'm hesitant to do it here because it's potentially going to be backported and we want to keep the scope small in those in order to not introduce regressions.